### PR TITLE
Upgrade Monitoring Documentation update

### PR DESCRIPTION
**Pull request by mrzta**
Upgrade Monitoring Documentation update

### CODE REVIEW

File: documentation/aks-upgrade.md [hunk 1]
```diff
@@ -105,3 +105,14 @@ az aks nodepool get-upgrades --resource-group <ResourceGroup> --cluster-name <Cl
 1. Follow the same manual process to upgrade the production cluster
     - Test with the welcome app: https://www.teacherservices.cloud/
     - Raise PR
+## Monitoring upgrade
```
Comment (saliceti) [inline documentation/aks-upgrade.md:108]: This is useful during the upgrade. Can it be added to "Upgrade process"? Alongside:
Continuously monitor it during the upgrade and check for downtime. Example:

File: documentation/Ingress-controller-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Ingress Controller Upgrade
+
+**Pre-requisits**
```
Comment (saliceti) [inline documentation/Ingress-controller-upgrade.md:3]: Typos: prerequisites

File: documentation/Ingress-controller-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+- AKS Cluster with the ingress controller you wish to upgrade
+- An app deployed to the cluster to monitor i.e. apply, register etc
+- Set you AKS context for the cluster `az aks get-credentials ...`
+
+**Monitoring the upgrade**
```
Comment (saliceti) [inline documentation/Ingress-controller-upgrade.md:8]: Please use `##` subsections for consistency

File: documentation/Ingress-controller-upgrade.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# Ingress Controller Upgrade
+
+**Pre-requisits**
+- AKS Cluster with the ingress controller you wish to upgrade
+- An app deployed to the cluster to monitor i.e. apply, register etc
+- Set you AKS context for the cluster `az aks get-credentials ...`
```
Comment (saliceti) [inline documentation/Ingress-controller-upgrade.md:6]: Use the `make ... get-cluster-credentials` instead (link to https://github.com/DFE-Digital/teacher-services-cloud#kubectl)

File: documentation/Ingress-controller-upgrade.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# Ingress Controller Upgrade
+
+**Pre-requisits**
+- AKS Cluster with the ingress controller you wish to upgrade
+- An app deployed to the cluster to monitor i.e. apply, register etc
```
Comment (saliceti) [inline documentation/Ingress-controller-upgrade.md:5]: The welcome apps can be used as they are always deployed: https://test.teacherservices.cloud/, https://platform-test.teacherservices.cloud/ etc

File: documentation/Ingress-controller-upgrade.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+- Set you AKS context for the cluster `az aks get-credentials ...`
+
+**Monitoring the upgrade**
+- Start the monitoring in a separate window using curl. This will monitor any app downtime please open a new window or terminal session for this and suite the url your current app deployment
```
Comment (saliceti) [inline documentation/Ingress-controller-upgrade.md:9]: Repetition: "in a separate window" and "please open a new window or terminal session for this"

File: documentation/Ingress-controller-upgrade.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  ```
+  while true; do date; curl -si https://www.apply-for-teacher-training.service.gov.uk/candidate/account | grep HTTP; sleep 1; done
```
Comment (saliceti) [inline documentation/Ingress-controller-upgrade.md:11]: Update with the welcome app URLs?

File: documentation/Ingress-controller-upgrade.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+  ```
+- Open another terminal session and paste the below command
+  ```
+  kubectl get pods -n default -w
+  ```
+**Upgrade**
+
+- To start the upgrade Variable called `ingress_nginx_version` located in `cluster/terraform_kubernetes/config/development.tfvars.json` which is developement cluster specific should be changed
```
Comment (saliceti) [inline documentation/Ingress-controller-upgrade.md:20]: Please rephrase

File: documentation/Ingress-controller-upgrade.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+- This is an opotunity to test changes in your developement cluster before rolling out to higher environment test, prod
```
Comment (saliceti) [inline documentation/Ingress-controller-upgrade.md:21]: opportunity
development

File: documentation/Ingress-controller-upgrade.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+- Set the Value of the variable i.e. `"ingress_nginx_version": "4.7.1"`
```
Comment (saliceti) [inline documentation/Ingress-controller-upgrade.md:22]: value

File: documentation/Ingress-controller-upgrade.md [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+- Run terraform apply i.e change cluster3 to your developement cluster
+
+```
+make development terraform-apply ENVIRONMENT=cluster3
+```
+- At this point monitor your cluster as well as curl response and make sure it return 200 as per the Monitoring
+- You will also see the pod recreated through its life cucle below
+
+        ```
+        apply-redis-review-1234-5f6588767c-6jkpb 0/1 Pending 0 0s
```
Comment (saliceti) [inline documentation/Ingress-controller-upgrade.md:32]: This should show the infgress-nginx pods

File: documentation/Ingress-controller-upgrade.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+- To start the upgrade open the file `test.tfvars.json` located at `cluster/terraform_kubernetes/config`
+- This is an oportunity to test changes in test cluster before rolling out to higher environment like prod
+- Set the value of the variable i.e. `"ingress_nginx_version": "4.7.1"`
+- Run terraform apply
+
+```
+make test terraform-apply CONFIRM_TEST=yes
+```
+- At this point monitor your cluster as well as curl response and make sure it return `200` .
+- You will also see the pod recreated through its life cucle below:
```
Comment (smneal) [inline documentation/Ingress-controller-upgrade.md:28]: `returns` and `cycle`

File: documentation/Ingress-controller-upgrade.md [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+        ```
+        ingress-nginx-admission-create-kksm2        0/1     Pending   0          0s
+        ingress-nginx-admission-create-kksm2        0/1     ContainerCreating   0          0s
+        ingress-nginx-admission-create-kksm2        1/1     Running             0          3s
+        ingress-nginx-admission-create-kksm2        0/1     Completed           0          4s
+
+
+        ```
+- Follow the same process for other environemt production and raise a PR for each.
+- Please give atleast 24 hours notice to the application teams before upgrade.
```
Comment (smneal) [inline documentation/Ingress-controller-upgrade.md:39]: `environment`. The sentence needs rewording as it doesn't make sense.

`at least`

**Review by smneal — APPROVED** [review]

**Pull request merged**